lib/filewatch/read_mode/handlers/read_zip_file.rb

```diff
@@ -35,12 +35,23 @@ def handle_specifically(watched_file)
         watched_file.listener.deleted
         watched_file.unwatch
       ensure
-        buffered.close unless buffered.nil?
-        decoder.close unless decoder.nil?
-        gzip_stream.close unless gzip_stream.nil?
-        file_stream.close unless file_stream.nil?
+        # rescue each close individually so all close attempts are tried
+        close_java_closeable(buffered) unless buffered.nil?
+        close_java_closeable(decoder) unless decoder.nil?
+        close_java_closeable(gzip_stream) unless gzip_stream.nil?
+        close_java_closeable(file_stream) unless file_stream.nil?
       end
       sincedb_collection.unset_watched_file(watched_file)
     end
+
+    private
+
+    def close_java_closeable(closeable)
```

Comment:
or maybe `close_and_ignore` or something like that? because this method does 2 things, it closes but also explicitly ignores exceptions.